### PR TITLE
Fixup: Add ambiguous month and day to date annotations

### PR DESCRIPTION
**Pull request by kimandrews**
## Description of proposed changes

This PR fixes date annotations for samples in `ingest/defaults/annotations.tsv` for which only the year is known. This is accomplished by adding ambiguous month and day to the date.

## Checklist

- [ ] Checks pass

### CODE REVIEW

**Review by victorlin — APPROVED** [review]

**Review by genehack — APPROVED** [review]

File: ingest/defaults/annotations.tsv [hunk 1]
```diff
@@ -9,26 +9,26 @@
 # sequences of measles virus strains in the Edmonston vaccine lineage
 # https://doi.org/10.1128/jvi.75.2.910-920.2001
 AF266288	strain	Measles strain Edmonston WT
-AF266288	date	1954
+AF266288	date	1954-XX-XX
```
Comment (jameshadfield) [inline ingest/defaults/annotations.tsv:12]: Which tools need this ambiguous coding? 

Within the curate pipeline `merge-user-metadata` is after `format-dates` (which would do this conversion automatically). I'm somewhat surprised that `augur refine` / `augur filter` etc don't handle a plain year, which is what this PR implies...

Comment (joverlee521) [reply ingest/defaults/annotations.tsv:12]: This is mostly to keep dates consistently in the ISO 8601 date format in our final metadata.tsv
It doesn't cause an error now, but based on discussion in https://github.com/nextstrain/augur/issues/882, I think we want to ensure that all dates are correctly formatted before running downstream augur commands.

**Review by joverlee521 — APPROVED** [review]

File: ingest/defaults/annotations.tsv (hunk 1 above)
Comment (jameshadfield) [reply ingest/defaults/annotations.tsv:12]: Thanks for the context @joverlee521

> It doesn't cause an error now

Are `YYYY` dates currently treated as `YYYY-XX-XX` within those downstream functions currently?

Comment (joverlee521) [reply ingest/defaults/annotations.tsv:12]: I believe all dates get transformed into numerical dates through `get_numerical_date_from_value` which does [assume year only dates are `YYYY-XX-XX`](https://github.com/nextstrain/augur/blob/556e999e769f9c791aa6760cf4aafad1003a5df0/augur/dates/__init__.py#L115-L117)

**Pull request merged**